src/apis/tenderlyapi.py

```diff
@@ -0,0 +1,95 @@
+"""
+TenderlyAPI for simulating transactions on tenderly.
+"""
+# pylint: disable=logging-fstring-interpolation
+
+from os import getenv
+from typing import Any, Optional
+import requests
+from dotenv import load_dotenv
+from src.helper_functions import get_logger
+from src.constants import (
+    SETTLEMENT_CONTRACT_ADDRESS,
+    REQUEST_TIMEOUT,
+)
+
+
+class TenderlyAPI:
+    """
+    Class for simulating transactions on tenderly.
+    """
+
+    def __init__(self) -> None:
+        self.logger = get_logger()
+        load_dotenv()
+        self.tenderly_url = (
+            "https://api.tenderly.co/api/v1/account/"
+            + str(getenv("TENDERLY_USER"))
+            + "/project/"
+            + str(getenv("TENDERLY_PROJECT"))
+            + "/simulate"
+        )
+
+    def simulate_solution(
+        self, solution: dict[str, Any], block_number: int, internalize: bool = True
+    ) -> Optional[dict[str, Any]]:
+        """Simulate a solution from the solver competition at a specified block number.
+        The solution dictionary follows the convention from the competition endpoint.
+        If internalize == False, uninternalized calldata is used for the simulation if it exists.
+        """
+        if not internalize and "uninternalizedCallData" in solution:
+            calldata = solution["uninternalizedCallData"]
+        else:
+            calldata = solution["callData"]
+
+        return self.simulate_transaction(
+            calldata, solution["solverAddress"], block_number
+        )
+
+    def simulate_transaction(
+        self, calldata: str, solver: str, block_number: int
+    ) -> Optional[dict[str, Any]]:
+        """Simulate a transaction from solver with given calldata at a specified block number."""
```

Comment:
At first, the solution/transaction usage confused me. Should we maybe just name this `simulate_calldata` ?

Reply:
I changed the name to `simulate_calldata`